packages/cli/src/commands/setup/vite/viteHandler.js

```diff
@@ -0,0 +1,136 @@
+import c from '../../../lib/colors'
+import { isTypeScriptProject } from '../../../lib/project'
+
+const { version } = JSON.parse(
+  fs.readFileSync(path.resolve(__dirname, '../../../../package.json'), 'utf-8')
+)
+
+export const handler = async ({ force, verbose, addPackage }) => {
+  const ts = isTypeScriptProject()
+  const tasks = new Listr(
+    [
+      {
+        title: 'Confirmation',
+        task: async (_ctx, task) => {
+          const confirmation = await task.prompt({
+            type: 'Confirm',
+            message: 'Vite support is experimental. Continue?',
+          })
+
+          if (!confirmation) {
+            throw new Error('User aborted')
+          }
+        },
+      },
+      {
+        title: 'Adding vite.config.js...',
+        task: () => {
+          const viteConfigPath = `${getPaths().web.base}/vite.config.${
+            ts ? 'ts' : 'js'
+          }`
+
+          const templateContent = fs.readFileSync(
+            path.resolve(__dirname, 'templates', 'vite.config.ts.template'),
+            'utf-8'
+          )
+
+          const viteConfigContent = ts
+            ? templateContent
+            : transformTSToJS(viteConfigPath, templateContent)
+
+          return writeFile(viteConfigPath, viteConfigContent, {
+            overwriteExisting: force,
+          })
+        },
+      },
+      {
+        title: 'Adding Vite bundler flag to redwood.toml...',
+        task: (_ctx, task) => {
+          const redwoodTomlPath = getConfigPath()
+          const configContent = fs.readFileSync(redwoodTomlPath, 'utf-8')
+
+          if (!configContent.includes('bundler = vite')) {
+            // Use string replace to preserve comments and formatting
+            writeFile(
+              redwoodTomlPath,
+              configContent.replace('[web]', '[web]\n  bundler = "vite"'),
+              {
+                overwriteExisting: true, // redwood.toml always exists
+              }
+            )
+          } else {
+            task.skip('Vite bundler flag already set in redwood.toml')
+          }
+        },
+      },
+      {
+        title: 'Creating new entry point in `web/src/entry-client.jsx`...',
+        task: () => {
+          // Keep it as JSX for now
+          const entryPointFile = path.join(
+            getPaths().web.src,
+            `entry-client.jsx`
+          )
+          const content = fs
+            .readFileSync(
+              path.join(
+                getPaths().base,
+                // NOTE we're copying over the index.js before babel transform
+                'node_modules/@redwoodjs/web/src/entry/index.js'
+              ),
+              'utf-8'
+            )
+            .replace('~redwood-app-root', './App')
+
+          return writeFile(entryPointFile, content, {
+            overwriteExisting: force,
+          })
+        },
+      },
+      {
+        ...addWebPackages([`@redwoodjs/vite@${version}}`]),
```

Comment:
Here's the second part; and here's the corresponding setup auth0 code: https://github.com/redwoodjs/redwood/blob/a3e9480d2ae53afe1665d608cd7a49fd025f9fe0/packages/auth-providers/auth0/setup/src/setupHandler.ts#L20-L23